### PR TITLE
Adds in new search,list svg icons and blacklight_icons helper

### LINKED ISSUE #1552: Search submit button styling is off on mobile view
Using Galaxy S5 setting on Google Chrome. Search submit button is too small and has lost its text.

demo projectblacklight org- galaxy s5

### PR DESCRIPTION
**Pull request by mejackreed**
This commit adds in several things:

 - Blacklight::Icons - a svg icon reader class
 - Blacklight::IconHelperBehavior - a helper that calls and wraps in a span,
   also caches the svg lookups
 - svg icons for search, and list. Both optimized for size reduction.

Link to sketch document used to create svgs: https://drive.google.com/open?id=0BzWuWHFTTIPES0tXUnkwQXZNaFk

screen shot 2016-11-03 at 9 49 48 pm
screen shot 2016-11-03 at 9 49 37 pm

Closes #1552

### CODE REVIEW

File: app/assets/stylesheets/blacklight/_icons.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+.btn-warning {
+  .blacklight-icons g {
+    stroke: $btn-warning-color;
+  }
+}
+
+.btn-danger {
+  .blacklight-icons g {
```
Comment (mejackreed) [inline app/assets/stylesheets/blacklight/_icons.scss]: I think we have to do this unfortunately

File: app/helpers/blacklight/icon_helper_behavior.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# Module to help generate icon helpers for SVG images
+module Blacklight::IconHelperBehavior
+  ##
+  # Returns the raw SVG (String) for a Blacklight Icon located in 
+  # app/assets/images/blacklight/*.svg. Caches them so we don't have to look up
+  # the svg everytime.
+  # @param [String, Symbol] icon_name
+  # @return [String]
+  def blacklight_icon(icon_name, options = {})
+    Rails.cache.fetch("blacklight_icons/#{icon_name}") do
```
Comment (cbeer) [inline app/helpers/blacklight/icon_helper_behavior.rb]: Seems like this should vary with `options` too, yeah?

Comment (mejackreed) [inline app/helpers/blacklight/icon_helper_behavior.rb]: Agreed. Is `Rails.cache.fetch("blacklight_icons/#{icon_name}/{options.to_s}")` to hacky? Any other suggestions?

Comment (cbeer) [inline app/helpers/blacklight/icon_helper_behavior.rb]: Does `Rails.cache.fetch` take compound arguments, maybe `[:blacklight_icons, icon_name, options]`? I believe the cache helpers do, but I'm uncertain where that's implemented in the stack.

Failing that, something like `options.to_query` may be a little better?

**Comment by jrochkind** [comment]
You can easily extract (or google for an extraction) of individual glyphs from FontAwesome or other similar fonts, as svgs, if you like their illustrations.

**Pull request merged**